<launch>

	<machine
		name="upper_pi"
		address="upper_pi"
		user="upperpi"
		password="mcgreen"
		timeout="10"
		env-loader="~/upper_ws/devel/remote_env_loader.sh"
	/>

	<group ns="Sensors">
		<param name="rate" value="100" type="int"/>
		<param name="default_ultra" value="80" type="int"/>
		<param name="threshold_ultra" value="40" type="int" />

		<node machine="upper_pi" name="Right_Arm_Sensor" pkg="mcgreen_control" respawn="true" type="arm_sensor.py">
			<param name="topic" value="/right_arm_sensor"/>
		</node>

		<node machine="upper_pi" name="Left_Arm_Sensor" pkg="mcgreen_control" respawn="true" type="arm_sensor.py">
			<param name="topic" value="/left_arm_sensor"/>
		</node>
	</group>

		<node machine="upper_pi" name="Face_Controller" pkg="mcgreen_control" type="face_controller.py"/>

		<node machine="upper_pi" name="Head_Servo_Driver" pkg="mcgreen_control" type="head_servo_driver.py"/>

		<node machine="upper_pi" name="Right_Linear_Actuator" pkg="mcgreen_control" respawn="true" type="linear_actuator_driver.py">
			<param name="side" value="right"/>
		</node>

		<node machine="upper_pi" name="Left_Linear_Actuator" pkg="mcgreen_control" respawn="true" type="linear_actuator_driver.py">
			<param name="side" value="left"/>
		</node>


	<machine
		name="lower_pi"
		address="lower_pi"
		user="lowerpi"
		password="mcgreen"
		timeout="10"
		env-loader="~/lower_ws/devel/remote_env_loader.sh"
	/>

	<node machine="lower_pi" name="Drivetrain_Driver" pkg="mcgreen_control" type="drivetrain_driver.py"/>

	<node machine="lower_pi" name="OLED_Screen_Controller" pkg="mcgreen_control" type="oled_screen_controller.py">
		<param name="rate" value="10" type="int"/>
	</node>

	<node machine="lower_pi" name="Remote_Control_Reciever" pkg="mcgreen_control" type="remote_control_reciever.py">
		<param name="rate" value="20" type="int"/>
	</node>


	<node name="Remote_Sensors_Processor" pkg="mcgreen_control" respawn="true" type="remote_sensors_processor.py">
		<param name="rate" value="100" type="int"/>
	</node>

	<node name="Movement_Output" pkg="mcgreen_control" respawn="true" type="movement_output.py" >
		<param name="rate" value="50" type="int"/>
	</node>

	<node name="Mode_Select" pkg="mcgreen_control" respawn="true" type="mode_select.py">
		<param name="speed_multiplier" value="2"/>
		<param name="joystick_threshold" value="2"/>
	</node>

	<node name="Game_Interface" pkg="mcgreen_control" type="game_interface.py"/>

</launch>
